### ISSUE TITLE
Skip RUF039 when \n, \t, ... are involved

### ISSUE THREAD

**Issue by alexprengere** [astral-sh/ruff#14529]
The latest Ruff 039 rule is a bit noisy (see #14527 too), and I think there are false positives: running it on a private codebase, I had a lot of:

```
_RE_STRIP = re.compile("[ \t\r\n]+")
                        ^^^^^^^^^^ RUF039
= help: Replace with raw string
```
Unless I missed something, there is no *easy* way to convert that one to a raw string, as `r"\n"` is semantically different to `"\n"` (but I agree that is a good general rule, when possible).

**Comment by MichaReiser** [comment]
Converting to a raw string should be possible for all escape sequence except `\b`

![image](https://github.com/user-attachments/assets/0f6e2848-0156-4372-8ff7-d15c6a2c2450)
[source](https://docs.python.org/3/library/re.html#re.Pattern)

You can also verify this using the repl

```python
>>> re.match("\n", "\n")
<re.Match object; span=(0, 1), match='\n'>
>>> re.match(r"\n", "\n")
<re.Match object; span=(0, 1), match='\n'>
>>> 
```
